This patch fixes vendor SDK header "c_types.h" for C99 compatibility, while keeping non-C99 types.

--- a/include/c_types.h.orig
+++ b/include/c_types.h
@@ -6,17 +6,13 @@
 #ifndef _C_TYPES_H_
 #define _C_TYPES_H_
 
-typedef unsigned char       uint8_t;
-typedef signed char         sint8_t;
-typedef signed char         int8_t;
-typedef unsigned short      uint16_t;
-typedef signed short        sint16_t;
-typedef signed short        int16_t;
-typedef unsigned long       uint32_t;
-typedef signed long         sint32_t;
-typedef signed long         int32_t;
+#include <stdint.h>
+#include <stdbool.h>
+
+typedef int8_t              sint8_t;
+typedef int16_t             sint16_t;
+typedef int32_t             sint32_t;
 typedef signed long long    sint64_t;
-typedef unsigned long long  uint64_t;
 typedef unsigned long long  u_int64_t;
 typedef float               real32_t;
 typedef double              real64_t;
@@ -80,10 +76,7 @@
 #endif /* ICACHE_FLASH */
 
 #ifndef __cplusplus
-typedef unsigned char   bool;
 #define BOOL            bool
-#define true            (1)
-#define false           (0)
 #define TRUE            true
 #define FALSE           false
 
